### PR TITLE
API endpoints

### PR DESCRIPTION
**Pull request by Lucas-Masaba**
# Project Requirements ⚙️ 
- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post.
- API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/api/api_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      @posts = Post.where(user_id: @user.id)
+      render json: @posts
+    end
+
+    def list_comments
+      @user = User.find_by_api_token(request.headers['X-AUTH-TOKEN'])
+      @comments = Comment.where(user_id: @user.id)
+      render json: @comments
+    end
+
```
Comment (andersonlebon) [inline app/controllers/api/api_controller.rb]: - [x] Kindly note that this should return comments for the given post not all comments for the current user. Hint :point_down: 

### Example
```rb
@comments = Comment.where(user_id: @user.id, post_id: params[:post_id])
```

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Lucas-Masaba 

Good job so far! I know you have worked hard on making this milestone There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [x] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/api_controller.rb (hunk 1 above)
Comment (Lucas-Masaba) [reply app/controllers/api/api_controller.rb]: Thank you for the suggestion 👍 .  I have implemented the changes

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾🎉✔**
Hello @Lucas-Masaba and @TheLuisAbarca 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally. ✔
- Github inline comments feature is used properly.
- You asked for the changes in a polite and friendly way


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**